modules/apps/headless/headless-user-notification/headless-user-notification-test/src/testFunctional/macros/UserNotificationAPI.macro

```diff
@@ -0,0 +1,129 @@
+definition {
+
+	macro _curlNotificationTemplates {
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (isSet(notificationTemplateId)) {
+			var api = "notification-templates/${notificationTemplateId}";
+		}
+		else {
+			var api = "notification-templates";
+		}
+
+		var curl = '''
+			${portalURL}/o/notification/v1.0/${api} \
+				-u test@liferay.com:test \
+				-H accept: application/json \
+				-H 'Content-Type: application/json' \
+		''';
+
+		return ${curl};
+	}
+
+	macro _curlUserNotifications {
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (isSet(userNotificationId)) {
+			var api = "user-notifications/${userNotificationId}";
+
+			if (isSet(parameter)) {
+				var api = "user-notifications/${userNotificationId}/${parameter}";
+			}
+		}
+		else if (isSet(userAccountId)) {
+			var api = "user-accounts/${userAccountId}/user-notifications";
+		}
+		else {
+			var api = "my-user-notifications";
+		}
+
+		var curl = '''
+			${portalURL}/o/headless-user-notification/v1.0/${api} \
+				-H accept: application/json \
+				-u test@liferay.com:test \
+		''';
+
+		return ${curl};
+	}
+
+	macro createNotificationTemplate {
+		Variables.assertDefined(parameterList = "${name},${recipientType},${userScreenName},${en_US},${type}");
+
+		var curl = UserNotificationAPI._curlNotificationTemplates();
+		var body = '''
+			-d {
```

Comment:
The format seems strange to me. Could you try to update using TAB in -d{} and see if it can pass the sf? I mean something like this:
var body = '''
    -d {
        "editorType": "richTest",
        ....
        "type": "${type}"
    }
''';

Reply:
I used the TAB in -d{}, and ran the SF again

Reply:
The updates look good, thanks!